gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,389 @@
+      for (Member member : members) {
+        if (member.emailOrDomain() != null) {
+          membersPb.add(member.type().prefix() + member.emailOrDomain());
+        } else {
+          membersPb.add(member.type().prefix());
+        }
+      }
+      bindingPb.setMembers(membersPb);
+      return bindingPb;
+    }
+
+    static Binding fromPb(com.google.api.services.cloudresourcemanager.model.Binding bindingPb) {
+      RoleType role =
+          (bindingPb.getRole() == null)
+              ? null : RoleType.valueOf(bindingPb.getRole().split("/")[1].toUpperCase());
+      List<Member> members = new ArrayList<>();
+      if (bindingPb.getMembers() != null) {
+        for (String memberPb : bindingPb.getMembers()) {
+          String[] memberInfo = memberPb.split(":", 2);
+          String memberTypeStr = memberInfo[0];
+          String emailOrDomain = (memberInfo.length > 1) ? emailOrDomain = memberInfo[1] : null;
+          switch (memberTypeStr) {
+            case "allUsers":
+              members.add(new Member(MemberType.ALL_USERS, null));
+              break;
+            case "allAuthenticatedUsers":
+              members.add(new Member(MemberType.ALL_AUTHENTICATED_USERS, null));
+              break;
+            case "user":
+              members.add(new Member(MemberType.USER, checkNotNull(emailOrDomain)));
+              break;
+            case "serviceAccount":
+              members.add(new Member(MemberType.SERVICE_ACCOUNT, checkNotNull(emailOrDomain)));
+              break;
+            case "group":
+              members.add(new Member(MemberType.GROUP, checkNotNull(emailOrDomain)));
+              break;
+            case "domain":
+              members.add(new Member(MemberType.DOMAIN, checkNotNull(emailOrDomain)));
+              break;
+            default:
+              throw new UnsupportedOperationException("Unsupported member type: " + memberTypeStr);
+          }
+        }
+      }
+      return new Binding(role, members);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(role, members);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof Binding && Objects.equals(this.role, ((Binding) obj).role)
+          && Objects.equals(this.members, ((Binding) obj).members);
+    }
+  }
+
+  public static final class Builder {
+    private List<Binding> bindings;
+    private String etag;
+    private Integer version;
+
+    private Builder() {
+      bindings = new ArrayList<>();
+    }
+
+    public Builder addBinding(Binding binding) {
+      this.bindings.add(binding);
+      return this;
+    }
+
+    public Builder removeBinding(Binding binding) {
+      this.bindings.remove(binding);
+      return this;
+    }
+
+    public Builder clearBindings() {
+      this.bindings = new ArrayList<>();
+      return this;
+    }
+
+    public Builder bindings(List<Binding> bindings) {
+      this.bindings = checkNotNull(bindings);
```

Comment:
As for `Binding` I think here we should create a new `ArrayList` to avoid changes outside of the builder: `this.bindings = Lists.newArrayList(checkNotNull(bindings));` or `this.bindings = new ArrayList(checkNotNull(bindings));`